synfig-core/src/synfig/valuenodes/valuenode_animatedfile.cpp

```diff
@@ -130,21 +168,138 @@ class ValueNode_AnimatedFile::Parser
+				line_num++;
+
+				if (line.empty()) {
+					warning(_("TSV file (line %zu): skipping empty line"), line_num);
+					continue;
+				}
+
+				const auto tab_pos = line.find('\t');
+				if (tab_pos == std::string::npos) {
+					error(_("TSV file (line %zu): no TAB separator"), line_num);
+					continue;
+				}
+
+				double seconds = strtodouble(line.substr(0, tab_pos));
+
+				std::string data = line.substr(tab_pos+1);
+
+				Time time(seconds);
+				values[time] = data;
+			}
+		}  catch (...) {
+			return false;
+		}
+
+		return s.eof() || !s.fail();
+	}
+
+	/// Parse a Rhubarb XML file (lipsync software)
+	/// \param[in] s the XML stream to be parsed
+	/// \param[out] value the map of time -> layer name
+	/// \param[out] fields the map of metafield -> value
+	/// \return true if successful. There is no guarantee about stream offset in case of failure
+	static bool parse_xml(istream &s, map<Time, String> &values, map<String, String> &fields)
+	{
+		try {
+			xmlpp::DomParser parser;
+			parser.parse_stream(s);
+			if (parser) {
+				auto doc = parser.get_document();
+				if (!doc) {
+					return false;
+				}
+				auto root = doc->get_root_node();
+				if (!root) {
+					return false;
+				}
+				if (root->get_name() != "rhubarbResult") {
+					synfig::error(_("XML animated file: it only supports Rhubarb Lip Sync files. Found %s"), root->get_name().c_str());
+					return false;
+				}
+				if (auto metadata_node = root->get_first_child("metadata")) {
+					if (auto sound_node = metadata_node->get_first_child("soundFile")) {
+						if (auto sound_element = dynamic_cast<xmlpp::Element*>(sound_node)) {
+							if (auto sound_text_node = sound_element->get_child_text())
+								fields["sound"] = sound_text_node->get_content();
+						}
+						if (fields.count("sound"))
+							synfig::warning(_("XML animated file: wrong contents for Rhubarb metadata > soundFile node"));
+					}
+				}
+				if (auto mouthcues_node = root->get_first_child("mouthCues")) {
+					for (auto mouthcue_node : mouthcues_node->get_children("mouthCue")) {
+						if (auto mouthcue_element = dynamic_cast<xmlpp::Element*>(mouthcue_node)) {
+							if (auto mouthcue_text_node = mouthcue_element->get_child_text()) {
+								std::string phoneme = mouthcue_text_node->get_content();
+								if (phoneme.empty()) {
+									synfig::warning(_("XML animated file: missing phoneme for Rhubarb mouthcue element: using X"));
+									phoneme = 'X';
+								}
+
+								if (auto start_attribute = mouthcue_element->get_attribute("start")) {
+									double seconds = strtodouble(start_attribute->get_value());
+									Time time(seconds);
+									values[time] = phoneme;
+								} else {
+									synfig::warning(_("XML animated file: missing attribute 'start' for Rhubarb mouthcue element: skipping"));
+								}
+							}
+						}
+					}
+				}
+				return true;
+			}
+		} catch(xmlpp::internal_error &x) {
+			if (std::string(x.what()) == "Couldn't create parsing context")
```

Comment:
I think it's better to report a error even if it doesn't match the description. I mean:
```
if (std::string(x.what()) == "Couldn't create parsing context")
    synfig::error(_("XML animated file: Can't XML open file"));
else
    synfig::error(x.what());
```